server/images.go

```diff
@@ -1105,17 +1115,27 @@ func PullModel(ctx context.Context, name string, regOpts *RegistryOptions, fn fu
 	return nil
 }
 
-func pullModelManifest(ctx context.Context, mp ModelPath, regOpts *RegistryOptions) (*ManifestV2, error) {
+func pullModelManifest(ctx context.Context, mp ModelPath, currentDigest string, regOpts *RegistryOptions) (*ManifestV2, error) {
```

Comment:
This isn't how I imagined this working. It shouldn't be necessary to pass in the digest. If the manifest exists, it can calculate the digest and set `If-None-Match`. It will allow any pull request, not just an explicit upgrade, to reuse this logic